### PR TITLE
[Zephyr] Enabling IPC interrupt

### PR DESCRIPTION
**Pull request by dcpleung**
This implements a minimal glue layer to enable interrupts with Zephyr, and IPC interrupts are then enabled.

This includes everything in #1704.
Requires https://github.com/thesofproject/zephyr/pull/7 on the Zephyr side to work.

### CODE REVIEW

File: src/audio/dai.c [hunk 1]
```diff
@@ -74,7 +74,7 @@ static void dai_buffer_process(struct comp_dev *dev, struct dma_sg_elem *next)
 	struct dai_data *dd = comp_get_drvdata(dev);
```
Comment (dbaluta) [inline src/audio/dai.c]: @finikorg always explain in the commit message why the change is needed.

File: zephyr/include/arch/string.h [hunk 2]
```diff
@@ -43,8 +43,10 @@ static inline int arch_memcpy_s(void *dest, size_t dest_size,
 	if (!dest || !src)
 		return -EINVAL;
```
Comment (dbaluta) [inline zephyr/include/arch/string.h]: Please add the warning message inside the commit mesage. Also, isn't this code supposed to be part of initial submit?

Comment (Vudentz) [reply zephyr/include/arch/string.h]: @dbaluta this is just a poc exercise, it is not meant for master, at least not for now.

File: src/audio/dai.c (hunk 1 above)
Comment (finikorg) [reply src/audio/dai.c]: @dbaluta I have this in the commit message "Fixes issues using void * in arithmetic operations."

File: zephyr/include/arch/string.h (hunk 2 above)
Comment (dbaluta) [reply zephyr/include/arch/string.h]: @Vudentz all right :)

**Pull request merged**